### PR TITLE
created movie, session and authentication controllers and Dtos

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

File: src/main/java/com/cinema/config/AppConfig.java [hunk 1]
```diff
@@ -14,7 +14,8 @@
 @PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "com.cinema.dao",
-        "com.cinema.service"
+        "com.cinema.service",
+        "com.cinema.model.dto"
```
Comment (dariiabutorina) [inline src/main/java/com/cinema/config/AppConfig.java]: Is that necessary?
Take a look at your getSessionFactory method

File: src/main/java/com/cinema/model/dto/movie/MovieDtoMapper.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private final MovieService movieService;
+
+    public MovieDtoMapper(MovieService movieService) {
+        this.movieService = movieService;
+    }
+
+    public Movie getMovie(MovieRequestDto dto) {
+        Movie movie = new Movie();
+        movie.setTitle(dto.getTitle());
+        return movie;
```
Comment (dariiabutorina) [inline src/main/java/com/cinema/model/dto/movie/MovieDtoMapper.java]: What if we receive the full info about the movie from the FrontEnd-table consisting of all the movies?
Example : 
id______title______description
1 _____ Test 1 ____ Desc 1
2 _____ Test 2 ____ Desc 2

In my opinion, we should create the ready-to-work object or get from DB by the title, in such a case

File: src/main/java/com/cinema/config/AppConfig.java (hunk 1 above)
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/config/AppConfig.java]: do we need to build beans from models?

File: src/main/java/com/cinema/model/dto/session/MovieSessionResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.model.dto.session;
+
+import lombok.Data;
+
+@Data
+public class MovieSessionResponseDto {
+    private Long movieId;
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/dto/session/MovieSessionResponseDto.java:7]: let's add movieSessionId as well

File: src/main/java/com/cinema/model/dto/movie/MovieRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.cinema.model.dto.movie;
+
+import lombok.Data;
+
+@Data
+public class MovieRequestDto {
+    private String title;
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/dto/movie/MovieRequestDto.java:7]: what about description?

File: src/main/java/com/cinema/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @PostMapping
+    public RedirectView addMovieSession(@RequestBody MovieSessionRequestDto dto) {
+        movieSessionService.add(mapper.getMovieSession(dto));
+        return new RedirectView("/");
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllMovieSessions(Long id, String date) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/controller/MovieSessionController.java]: this is optional, but you can consider this approach as well 
@RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date

File: src/main/java/com/cinema/controller/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionDtoMapper mapper) {
+        this.movieSessionService = movieSessionService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping
+    public RedirectView addMovieSession(@RequestBody MovieSessionRequestDto dto) {
+        movieSessionService.add(mapper.getMovieSession(dto));
+        return new RedirectView("/");
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/controller/MovieSessionController.java]: let's remove all redirects, we don't need them

Comment (ilya449) [reply src/main/java/com/cinema/controller/MovieSessionController.java]: Removed

File: src/main/java/com/cinema/controller/MovieSessionController.java (hunk 5 above)
Comment (ilya449) [reply src/main/java/com/cinema/controller/MovieSessionController.java]: Ok, thank you

File: src/main/java/com/cinema/model/dto/movie/MovieRequestDto.java (hunk 4 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/dto/movie/MovieRequestDto.java:7]: Added

File: src/main/java/com/cinema/model/dto/session/MovieSessionResponseDto.java (hunk 3 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/dto/session/MovieSessionResponseDto.java:7]: Added

File: src/main/java/com/cinema/config/AppConfig.java (hunk 1 above)
Comment (ilya449) [reply src/main/java/com/cinema/config/AppConfig.java]: Fixed

Comment (ilya449) [reply src/main/java/com/cinema/config/AppConfig.java]: Isn't necessary, TY

File: src/main/java/com/cinema/model/dto/movie/MovieDtoMapper.java (hunk 2 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/dto/movie/MovieDtoMapper.java]: I've added ```description``` field and setting it in mapper. But getting movie from DB is also good idea

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's move all mappers to `service/mappers` folder

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**